C-Plus-Plus/cp/InversionCountProblem.cpp

```diff
@@ -0,0 +1,127 @@
+//Given an array, find the inversion count for the array.
+//Inversion count of an array:- It indicates how far the array is from being a sorted array. For a sorted array, it is zero.
+//Sorted Array:- An array in arranged in a particular order (for this question its acsending order)
+/*
+  For example:-
+  Input:- arr[] = {1, 5, 2, 9, 3}
+  Output:- 3
+  Inversions are:- 
+  (5, 2), (5, 3), (9, 3)
+  
+  Input:- arr[] = {1, 2, 3, 4, 5}
+  Ouput:- 0 
+*/
+
+/* 
+  Why we use Merge Sort approach?
+  If we will use simple approach, i.e., traverse through the array, and for every index, find the number of smaller elements on its right side of the array, using a nested loop and sum up the counts for all index in the array and print the sum, it will take O(n^2) Time Complexity.
+  Instead, we will use Merge Sort approach(Algorithm given below) to reduce Time Complexity to O(nlogn).
+*/  
```

Comment:
Rewrite it as,

```
/* 
  Why we use Merge Sort approach?
  If we will use simple approach, i.e., traverse through the array, and for every index, find the number of 
  smaller elements on its right side of the array, using a nested loop and sum up the counts for all index
  in the array and print the sum, it will take O(n^2) Time Complexity.
  Instead, we will use Merge Sort approach(Algorithm given below) to reduce Time Complexity to O(nlogn).
*/  
```